src/screens/Messages/List/ChatListItem.tsx

```diff
@@ -103,11 +104,16 @@ function ChatListItemReady({
     setShowActions(true)
   }, [])
 
-  const onPress = useCallback(() => {
+  const onPress = useCallback(async () => {
+    if (convo.unreadCount > 0) {
+      // We only increment the badge once per conversation, so even if there are multiple unreads
+      // we only want to decrement by one.
+      decrementBadgeCount(1)
```

Comment:
We're doing `markAsRead` within the `ConvoProvider`, could be a good spot for this too

Reply:
Yeah pry good to do there in case you foreground the app directly to the convo page

Reply:
Ahhh okay perfect let me adjust to that, wasn't sure if we did it or the server did it for us.